vendor/engines/saml_authentication/lib/saml_authentication/idp_entity_id_reader.rb

```diff
@@ -0,0 +1,44 @@
+module SamlAuthentication
+
+  # Overrides DeviseSamlAuthenticatable::DefaultIdpEntityIdReader.
+  #
+  # The default IdpEntityIdReader triggers an error on SLO because that request to
+  # us has a SAMLResponse parameter, which does not get parsed by OneLogin::RubySaml::Response
+  # correctly: "Issuer of the Response not found or multiple."
+  #
+  # Ideally, we will issue a PR to the devise_saml_authenticatable gem with these
+  # changes so it is corrected upstream.
+  class IdpEntityIdReader
+
+    def self.entity_id(params)
+      if params[:action] == "idp_sign_out" && params[:SAMLResponse]
+        OneLogin::RubySaml::Logoutresponse.new(
+          params[:SAMLResponse],
+          settings: Devise.saml_config,
+          allowed_clock_drift: Devise.allowed_clock_drift_in_seconds,
+        ).issuer
+      else
+        original_implementation(params)
```

Comment:
I’m not sure I quite follow the comment [here](https://github.com/tablexi/nucore-open/pull/1506/files#diff-74fdc61f331e681b5a8aa680f72296c6R25) and why this way of wrapping the original behavior improves clarity.

Perhaps I am missing something, but can’t we replace this line with a call to:

```
DeviseSamlAuthenticatable::DefaultIdpEntityIdReader.entity_id(params)
```

To me, that very clearly shows what we preempt and then fall back to, without a need to copy the method in this class, or rename it to `original_implementation`. We’re also not taking on an additional burden of maintaining the copy of `original_implementation` up to date with what the gem may change to upstream, and there is no need for comments to explain that we’re copying a method instead of just calling it.

Reply:
That's a fair question. I think I originally had this subclass `DefaultIdpEntityIdReader` and  call to `super`, but I felt that lost why we were needing to override it without needing to go into a `bundle open` to see what was missing. I also was feeling that with the override that I'd rather be in control of the whole thing and avoid them doing something under the hood that would break this.

I could go either way, though.